Add support for importing old FreeTube subscription.db files

T-vK opened the pull request:
If you want me to remove that workaround:

``` JavaScript
              if (importingOldFormat && profileObject.name === "All Channels") {
                const primaryProfile = JSON.parse(JSON.stringify(this.profileList[0]))
                primaryProfile.subscriptions = primaryProfile.subscriptions.concat(profileObject.subscriptions)
                this.updateProfile(primaryProfile)
              } else {
                this.updateProfile(profileObject)
```

or anything tell me.

VioletRose commented on src/renderer/components/data-settings/data-settings.js:
```diff
@@ -125,9 +125,43 @@ export default Vue.extend({
           textDecode.pop()
+          textDecode = textDecode.map(data => JSON.parse(data))
+
+          // TODO: Move this function to a better place
```
Is this TODO for future improvements, or something that you originally intended to handle before making the PR?

T-vK replied:
I'm just not familiar enough with the architecture to understand where a function like that would be best placed. So no I had no intention of moving the function, but if in order to get this PR accepted, it needs to be moved, I'm willing to do it. Someone just needs to tell me where exactly it needs to go.

PrestonN replied:
You are correct.  This function should be moved.  It can simply be moved towards the bottom of the methods in between `exportHistory` and `getChannelInfoInvidious`.  Within the `importFreeTubeSubscriptions` function, you would call it with the syntax `this.convertOldFreeTubeFormatToNew()` instead of the way it's called now.  You may need to include an `await` method as well to prevent the logic from being skipped due to it being asynchronous.

If you come across any issues with moving it or unsure of the syntax, please let me know.

PrestonN commented on src/renderer/components/data-settings/data-settings.js:
```diff
@@ -125,9 +125,43 @@ export default Vue.extend({
+                if (index === -1) { // profile doesn't exist yet
+                  convertedData.push({
+                    name: profile.value,
+                    bgColor: '#000000',
```
I understand defaulting this to black since the db file doesn't return a color.  Let's instead has it set to a random color.  I already have functions that can do this for you.  You will want `getRandomColor` and `calculateColorLuminance` from the utils VueX store to do this for you.  Let me know if you need help with the syntax as it might be strange to make these work if you're rather new to the framework.

PrestonN reviewed (changes requested):
Fix these issues and we're good.  If you have any questions with this then please let me know.

T-vK commented:
Alright, I fixed those issues. I also added:

``` JavaScript
// filter out subscriptions that already exist before concatenating
profileObject.subscriptions = profileObject.subscriptions.filter(newSub => {
  const existingSub = primaryProfile.subscriptions.find(existingSub => existingSub.id === newSub.id)
  return !existingSub // return false if sub already exists in default profile
})
```
Because otherwise the `All Channels` profile would end up with duplicates when importing the same subscriptions.db multiple times.

PrestonN reviewed (approved):
This looks good to me now.  Thank you for the contribution!
